### PR TITLE
Improve Track/Tracklist performance

### PR DESCRIPTION
**Pull request by LloydW93**
Also fixes permissions on a cold/no cache.

### CODE REVIEW

File: src/Classes/Database.php [hunk 1]
```diff
@@ -174,17 +173,19 @@ public function fetchAll($sql, $params = [])
...
      * @return Array The requested result row, or an empty array on failure
      * @throws MyRadioException
      */
     public function fetchOne($sql, $params = [])
     {
-        try {
-            $result = $this->query($sql, $params);
-        } catch (MyRadioException $e) {
-            return [];
+        if (!is_resource($sql)) {
```
Comment (AJDurant) [inline src/Classes/Database.php]: if it is a resource then it will try to return $result which is undefined?

File: src/Classes/Database.php [hunk 2]
```diff
@@ -174,20 +173,22 @@ public function fetchAll($sql, $params = [])
...
+            try {
+                $sql = $this->query($sql, $params);
+            } catch (MyRadioException $e) {
+                return [];
+            }
+        } else {
 
-        return pg_fetch_assoc($result);
+        return pg_fetch_assoc($sql);
```
Comment (AJDurant) [inline src/Classes/Database.php]: this is, perhaps, more silly...

File: src/Classes/ServiceAPI/MyRadio_TracklistItem.php [hunk 3]
```diff
@@ -227,9 +231,7 @@ private static function trackAmalgamator($result, $playlists = true)
 
             if ($playlists) {
                 $playlistobjs = iTones_Playlist::getPlaylistsWithTrack($trackobj);
-                foreach ($playlistobjs as $playlist) {
-                    $track['in_playlists'] .= $playlist->getTitle() . ', ';
-                }
+                $track['in_playlists'] = implode(', ', array_map(function($i){return $i->getTitle();}, $playlistobjs);
```
Comment (AJDurant) [inline src/Classes/ServiceAPI/MyRadio_TracklistItem.php]: Parse error: syntax error, unexpected ';' in src/Classes/ServiceAPI/MyRadio_TracklistItem.php on line 234

**Comment by LloydW93** [comment]
Much better.

**Comment by AJDurant** [comment]
you changed `myury.stats.jukeboxplaycounter.js` without changing it to `myradio`

**Comment by AJDurant** [comment]
you probably need to update the reference to that file, not just rename it...

**Comment by LloydW93** [comment]
Apparently I did but didn't include them in the commit...

**Pull request merged**